include/EVT/dev/storage/EEPROM.hpp

```diff
@@ -0,0 +1,119 @@
+#ifndef EVT_EEPROM_HPP
+#define EVT_EEPROM_HPP
+
+#include <stdint.h>
+
+namespace EVT::core::DEV {
+/**
+ * EEPROMs are a type of small non-volatile storage devices. This class
+ * represents features common across all EEPROMs.
+ */
+class EEPROM {
+
+public:
+    /**
+     * Read 8 bits of data from the given address.
+     *
+     * @param[in] address The address to read from
+     * @return The 8 bits of data at the address
+     */
+    virtual uint8_t readByte(uint32_t address) = 0;
+
+    /**
+     * Read 16 bits of data from the given address.
+     *
+     * @param[in] address The address to read from
+     * @return The 16 bits of data at the address
+     */
+    virtual uint16_t readHalfWord(uint32_t address) = 0;
+
+    /**
+     * Read 32 bits of data from the given address.
+     *
+     * @param[in] address The address to read from
+     * @return The 32 bits of data at the address
+     */
+    virtual uint32_t readWord(uint32_t address) = 0;
+
+    /**
+     * Read a number of consecutive bytes starting at a given memory address.
+     *
+     * @param[in] address The memory address to start reading from
+     * @param[in] numBytes The number of bytes to read
+     * @param[out] buffer Buffer to output bytes to
+     */
+    virtual void readBytes(uint8_t address, uint8_t numBytes, uint8_t* buffer) = 0;
```

Comment:
NIT: In similar usage (I2C) we have the format with the buffer parameter first then the number of bytes. It's a common practice in most code bases